tests/test_config_space_from_yaml.py

```diff
@@ -0,0 +1,362 @@
+            },
+            "use_crf": {
+                "param_type": "searchable",
+                "name": "use_crf",
+                "type": "categorical",
+                "choices": [True, False],
+            },
+        }
+    )
+    return config
+
+
+@pytest.fixture()
+def lightning_classification_config_dict(base_config_dict: Dict[str, Any]) -> Dict[str, Any]:
+    config = deepcopy(base_config_dict)
+    config["embedding_name_or_path"] = "allegro/herbert-base-case"
+    config["parameters"].update(
+        {
+            "max_seq_length": {
+                "param_type": "constant",
+                "name": "max_seq_length",
+                "value": 128,
+            },
+            "optimizer": {
+                "param_type": "searchable",
+                "name": "optimizer",
+                "type": "categorical",
+                "choices": ["Adam", "AdamW"],
+            },
+            "use_scheduler": {
+                "param_type": "searchable",
+                "name": "use_scheduler",
+                "type": "categorical",
+                "choices": [False, True],
+            },
+            "warmup_steps": {
+                "param_type": "searchable",
+                "name": "warmup_steps",
+                "type": "int_uniform",
+                "low": 0,
+                "high": 200,
+                "step": 10,
+            },
+            "adam_epsilon": {
+                "param_type": "searchable",
+                "name": "adam_epsilon",
+                "type": "uniform",
+                "low": 0.0,
+                "high": 0.1,
+            },
+            "weight_decay": {
+                "param_type": "searchable",
+                "name": "weight_decay",
+                "type": "uniform",
+                "low": 0.0,
+                "high": 0.1,
+            },
+            "finetune_last_n_layers": {
+                "param_type": "searchable",
+                "name": "finetune_last_n_layers",
+                "type": "categorical",
+                "choices": [-1, 0, 1, 3, 5, 7, 9],
+            },
+            "classifier_dropout": {
+                "param_type": "searchable",
+                "name": "classifier_dropout",
+                "type": "discrete_uniform",
+                "low": 0.0,
+                "high": 0.5,
+                "q": 0.05,
+            },
+        }
+    )
+    return config
+
+
+@pytest.fixture()
+def lightning_classification_wrong_param_config_dict(
+    lightning_classification_config_dict: Dict[str, Any]
+) -> Dict[str, Any]:
+    config = deepcopy(lightning_classification_config_dict)
+    config["parameters"].update(
+        {"label_all_tokens": {"param_type": "constant", "name": "label_all_tokens", "value": False}}
+    )
+    return config
+
+
+@pytest.fixture()
+def base_yaml_config_file_path(
+    output_path: "TemporaryDirectory[str]", base_config_dict: Dict[str, Any]
+) -> Path:
+    output_path = Path(output_path.name).joinpath("config.yml")
```

Comment:
might be flavour but with Path you can use /

Reply:
true but in general we use joinpath() in this project which I personally prefer too